repository/managers.py

```diff
@@ -1,17 +1,112 @@
 from repository.restable import RESTManager
 from repository import auth
 
+from external_accounts.utils import get_giles_document_details
+
+import requests
 
 class RepositoryManager(RESTManager):
-    def __init__(self, configuration, **kwargs):
+    def __init__(self, **kwargs):
         self.user = kwargs.get('user')
+        self.repository = kwargs.pop('repository')
         if self.user:
-            kwargs.update({'headers': auth.jars_github_auth(self.user)})
-        super(RepositoryManager, self).__init__(configuration, **kwargs)
+            kwargs.update({'headers': auth.citesphere_auth(self.user)})
+        super(RepositoryManager, self).__init__(**kwargs)
 
     def get_raw(self, target, **params):
-        import requests
         headers = {}
         if self.user:
-            headers = auth.jars_github_auth(self.user)
+            headers = auth.citesphere_auth(self.user)
         return requests.get(target, headers=headers, params=params).content
+
+    def groups(self):
+        """Fetch Groups from the repository's endpoint"""
+        headers = auth.citesphere_auth(self.user)
+        url = f"{self.repository.endpoint}/api/v1/groups/"  # Use repository's endpoint
```

Comment:
you probably want the path in the settings file. or at least in a different class specifically for v1 of the api. If/when we move to a version 2 of the api, we want this to be fairly easily exchangable. thinking about this though, I'll create a new ticket for that!